xml/hardening.xml

```diff
@@ -3445,6 +3445,311 @@ Account Expires: Never</screen>
+     </listitem>
+     <listitem>
+      <para>
+       This will include the <literal>pam_securetty</literal> PAM module during the
+       authentication process on local consoles to allow root only to login on
+       tty devices that are listed in the file <filename>/etc/securetty</filename>.
+       Thus to actually limit the tty devices for root you need to remove all
+       entries from this file except one:
+      </para>
+     <screen>#
+# This file contains the device names of tty lines (one per line,
+# without leading /dev/) on which root is allowed to login.
+#
+tty1</screen>
+     </listitem>
+     <listitem>
+      <para>
+       After these changes root logins from non-allowed terminals will be
+       rejected right away without even querying the root password.
+      </para>
+     </listitem>
+    </itemizedlist>
+
+    <important>
+     <para>
+      One might be tempted to add the <literal>pam_securetty</literal> PAM
+      module to the <filename>/etc/pam.d/common-auth</filename> file to make it
+      effective for all PAM services. This would break <literal>su</literal>
+      and <literal>sudo</literal>, however, because these tools would then
+      also reject root authentications.
+     </para>
+    </important>
+
+    <important>
+     <para>
+      These configuration changes will also cause root logins on serial
+      consoles like <filename>/dev/ttyS0</filename> to be denied. In case you
+      require such use cases you need to list the respective tty devices
+      additionally in the <filename>/etc/securetty</filename> file.
+     </para>
+    </important>
+
+    <note>
+     <para>
+      The steps shown here are tailored towards PC architectures (&x86; and
+      &x86-64;). On architectures like &power; different terminal
+      device names than <literal>tty1</literal> may be used. Be careful not to
+      lock yourself out completely by specifying wrong terminal device names.
+      You can determine the device name of the terminal you are currently
+      logged into by running the <literal>tty</literal> command. Be careful
+      not to do this in a virtual terminal like via SSH or in a graphical
+      session (device names <filename>/dev/pts/&lt;N&gt;</filename>) but only
+      from an actual login terminal reachable via
+      <keycombo><keycap function="alt"/><keycap>F&lt;N&gt;</keycap></keycombo>.
+     </para>
+    </note>
+   </sect2>
+
+   <sect2 xml:id="sec-sec-prot-retrict-root-graphical">
+    <title>Restricting Graphical Session Logins</title>
+
+    <para>
+     On security sensitive systems like servers it is best not to run a
+     graphical user environment at all. If you require a graphical login
+     anyways then it is even more so important that the &rootuser;
+     user cannot login in a graphical session. Graphical programs are often
+     not designed to be run as &rootuser; and are more likely to
+     contain security issues than console programs.
```

Comment:
"On security sensitive systems..." same here. I'd took the liberty here to rephrase it a bit more.

```suggestion
    To improve security on your server, avoid to use
    graphical user environments at all.
    Graphical programs are often not designed to be run as
    &rootuser; and are more likely to contain security issues than
    console programs.
    If you require a graphical login, use a non-&rootuser; login.
    Configure your system to disallow &rootuser; login on graphical
    sessions.
```

I've structured it as follows: what is the goal ("to improve security"), then an explanation (the reason), and after that how to avoid it.